source/_posts/2017-02-07-falsehoods-programmers-believe-about-language.markdown

```diff
@@ -0,0 +1,189 @@
+being _comfortable_ using an application. If two competing applications exist
+and one is in a user's native language, they are very likely to pick that one
+even if it doesn't have all the bells and whistles.
+
+
+## Translating a single word of English is simple.
+
+Think of all the places in software where just a single word appears. If a
+translator is only presented with a YAML file of strings, how are they supposed
+to know whether it's "Back" as in "go back to where you were before" or as in
+"not the front"? Whether it's "Post" as in "post my comment, now" or as in
+"mail"? This can lead to some very confusing experiences for non-English users.
+
+Also, if we consider just verbs, often tenses collide in English. "Eat" appears
+in all of: "I will eat this" (future), "I eat this (often)" (habitual present),
+"Did you eat this" (past) and "I was going to eat that" (conditional perfect, I
+think. Who even names these things?!?).
+
+If we consider just _nouns_, some lanuages have what's called a case system
+which is used instead of word order and/or pronouns to denote a noun's
+relationship to the rest of the words in a sentence. So "gift" could have a
+different shape in translations of these sentences:
+
+* The gift is ready.
+* I gave the gift to you.
+* I threw a ball at the gift.
+* The gift's wrapping is pretty.
+* This is my gift.
+
+
+## Short words in English are short in other languages.
+
+If your GUI design has a small buttong for "Post" it is very probable that some
+other language will have a longer word for it (compare just the English
+"Publish"). This isn't just a feature of languages with smallish speaking
+communities. German is _notorious_ for forming new words by making compounds and
+they can get quite long compared to their English conterparts. E.g.
+"Nacktschnecke" (slug) or "Fledermaus" (bat).
+
+There are a _few_ words that are short in almost all languages they appear in:
+"the", "a", "mom", "dad", "I", "you", and things like that. These are words so
+commonly used and so old that they've been whittled down over time for
+efficiency's sake.  But most words that are short in one language are short out
+of pure happenstance and some languages (English is one) seem to have a
+preference for more shorter words while others prefer fewer longer words.
+
+
+## Translating a whole sentence of English is simple.
+
+A whole sentence as the smallest unit to be translated is better, but there is
+still plenty of room for ambiguity. Consider any sentence with "it" in it.
+In languages where every (or even some) noun is gendered, a translator needs to
+know what the antecedent is or at least its gender in order to correcly
+translate this word. If that antecedent is in a previous sentence, they'll have
+to guess and if they're wrong, the sentence will read as if it were written by a
+young child still learning the basics of grammar.
+
+
+## Templating sentences adds only a little complexity.
```

Comment:
This heading isn't really the falsehood, more like "If I provide a templated string to the translator, they can take it from there"